File: pdu/python/ev3_msgs/pdu_cdr_conv_Ev3PduTouchSensor.py
from .pdu_pytype_Ev3PduTouchSensor import Ev3PduTouchSensor
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_Ev3PduTouchSensor(writer: CdrWriter, src: Ev3PduTouchSensor):
    writer.write_uint32(src.value)


def cdr_body_to_py_Ev3PduTouchSensor(reader: CdrReader, dst: Ev3PduTouchSensor):
    dst.value = reader.read_uint32()
    return dst


def py_to_cdr_Ev3PduTouchSensor(src: Ev3PduTouchSensor) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Ev3PduTouchSensor(writer, src)
    return writer.bytes()


def cdr_to_py_Ev3PduTouchSensor(cdr_payload) -> Ev3PduTouchSensor:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Ev3PduTouchSensor()
    return cdr_body_to_py_Ev3PduTouchSensor(reader, dst)

File: pdu/javascript/ev3_msgs/pdu_cdr_conv_Ev3PduTouchSensor.js
import { Ev3PduTouchSensor } from './pdu_jstype_Ev3PduTouchSensor.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduEv3PduTouchSensorConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Ev3PduTouchSensor } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint32(src.value);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Ev3PduTouchSensor } dst
     * @returns { Ev3PduTouchSensor }
     */
    static cdr_body_to_js(reader, dst) {
        dst.value = reader.read_uint32();
        return dst;
    }

    /**
     * @param { Ev3PduTouchSensor } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Ev3PduTouchSensor }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Ev3PduTouchSensor());
    }
}
